### PR TITLE
[Merged by Bors] - Add Commands::new_from_entities

### PR DESCRIPTION
**Pull request by TheRawMeatball**
This change allows for creating `Commands` objects from just an entities reference, which allows for creating multiple dynamically in a normal system.

Context: https://discord.com/channels/691052431525675048/774027865020039209/960857605943726142

### CODE REVIEW

**Review by james7132 — APPROVED** [review]

**Review by james7132 — APPROVED** [review]

**Review by BoxyUwU — APPROVED** [review]

File: crates/bevy_ecs/src/system/commands/mod.rs [hunk 1]
```diff
@@ -48,6 +48,11 @@ impl<'w, 's> Commands<'w, 's> {
         }
     }
 
+    /// Create a new `Commands` from a queue and an [`Entities`] reference.
```
Comment (DJMcNab) [inline crates/bevy_ecs/src/system/commands/mod.rs:51]: This should warn that if the commands never get applied, they will never get applied.

I would not be surprised if we had some users not recognising this - at least then we can point them to these docs.

What is the expected usage pattern of this anyway?

**Review by DJMcNab — APPROVED** [review]
I don't think this is bad per se, just challenging to use correctly.

File: crates/bevy_ecs/src/system/commands/mod.rs (hunk 1 above)
Comment (TheRawMeatball) [reply crates/bevy_ecs/src/system/commands/mod.rs:51]: The expected usage is creating multiple `Commands` instances within a `par_for_each` context. I plan on making a wrapper api as well, so most users won't need to use this directly. As for changing the documentation, I'd argue it's not that necessary given the docs on `CommandQueue` should cover it.

**Pull request closed without merging**